leshan-core/src/main/java/org/eclipse/leshan/core/util/base64/DefaultBase64Decoder.java

```diff
@@ -39,8 +41,6 @@ public String decode(byte[] encoded) {
         if (!this.withoutPadding) {
```

Comment:
you still create a new decoder on decode.

Reply:
I don't understand where do i create new decoder on decode?

Reply:
Comment review seems to not be attached to the right line, so this doesn't help to understand :grimacing: 

In this class, there is 2 `decode` methods.
1 method reuses decoder created in constructor, the other recreate a new decoder on each `decode` call.

Decoder should be create only once in constructor for both methods.

Tell me if this is not clearer enough.

Reply:
As far as i can tell the only new instances are two new strings that are created in this method. I can't see where in this method could new decoder be created 🤔
```java
    @Override
    public String decode(byte[] encoded) {
        if (!this.withoutPadding) {
            validateEncodedData(new String(encoded));
        }
        byte[] decoded = decoder.decode(encoded);
        return new String(decoded);
    }
```

Reply:
This is in the other method. 
```diff
public byte[] decode(String encoded) {
        if (!this.withoutPadding) {
            validateEncodedData(encoded);
        }
--        Base64.Decoder decoder = urlSafe ? Base64.getUrlDecoder() : Base64.getDecoder();
        return decoder.decode(encoded);
}
```
(note that I was thinking that  `Base64.getUrlDecoder()` and` Base64.getDecoder() ` created a new decoder but this is a static reference. Anyway, I think it's more consistent to reuse the attribute create in constructor.

Maybe could even be replaced by : 
```diff
public byte[] decode(String encoded) {
--        if (!this.withoutPadding) {
--            validateEncodedData(encoded);
--        }
--       Base64.Decoder decoder = urlSafe ? Base64.getUrlDecoder() : Base64.getDecoder();
--        return decoder.decode(encoded);
++   return decode(encoded.getByte());
}
```

Not directly linked but I just see that Base64.decode(String) consider that String is encoded in ISO_8859_1 which is very strange to me. I will try to understand why...